File: bestWorstLotto/src/main/kotlin/bestWorstLotto.kt
class Solution {
    fun solution(lottos: IntArray, win_nums: IntArray): IntArray {
        var answer: IntArray = intArrayOf()
        var num0: Int = 0
        for(i in lottos.indices){
            if(lottos[i]==0) num0++
        }

        /*
        var countArray: BooleanArray = BooleanArray(6)
        for(i in lottos.indices){
            for(j in win_nums.indices){
                if(lottos[i] == win_nums[j]){
                   countArray[i] = true
                }
            }
        }
        var count: Int = 0
        for(cnt in countArray){
            if(cnt) count++
        }
        */

        var countArray: IntArray = IntArray(6)
        for(i in lottos.indices){
            for(j in win_nums.indices){
                if(lottos[i] == win_nums[j]) countArray[i]++
            }
        }
        var count: Int = countArray.sum()

        answer += rankAnswer(num0 + count)
        answer += rankAnswer(count)

        return answer
    }

    fun rankAnswer(rank: Int): Int{
        return when(rank){
            6 -> 1
            5 -> 2
            4 -> 3
            3 -> 4
            2 -> 5
            else -> 6
        }
    }
}

File: bestWorstLotto/src/main/kotlin/bestWorstLotto.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    vector<int> answer;


    int count=0;
    int max=0, min=0;
    for(int i = 0 ; i < lottos.size();i++)
        if(lottos.at(i)==0)
            max++;

    for(int i = 0 ; i < lottos.size();i++)
        for(int j = 0 ; j < win_nums.size();j++)
            if(lottos.at(i)==win_nums.at(j))
                count++;

    max += count;
    min += count;

    switch(max)
    {
        default: answer.push_back(6); break;
        case 2: answer.push_back(5); break;
        case 3: answer.push_back(4); break;
        case 4: answer.push_back(3); break;
        case 5: answer.push_back(2); break;
        case 6: answer.push_back(1); break;
    }
        switch(min)
    {
        default: answer.push_back(6); break;
        case 2: answer.push_back(5); break;
        case 3: answer.push_back(4); break;
        case 4: answer.push_back(3); break;
        case 5: answer.push_back(2); break;
        case 6: answer.push_back(1); break;
    }


    //answer.push_back(max);
    //answer.push_back(min);
    return answer;
}